backend/service/k8s/clientset.go

```diff
@@ -64,10 +69,24 @@ func newClientsetManager(rules *clientcmd.ClientConfigLoadingRules) (ClientsetMa
 		if err != nil {
 			return nil, err
 		}
-
 		lookup[name] = &ctxClientsetImpl{Interface: clientset, namespace: ns, cluster: ctxInfo.Cluster}
 	}
 
+	// If there is no configured cluster produced fallback to InClusterConfig
+	if len(lookup) == 0 {
+		restConfig, err := rest.InClusterConfig()
+		if err != nil {
+			return nil, err
+		}
+
+		clientset, err := k8s.NewForConfig(restConfig)
+		if err != nil {
+			return nil, fmt.Errorf("could not create k8s InClusterConfig: %w", err)
+		}
+
+		lookup[localCluster] = &ctxClientsetImpl{Interface: clientset, namespace: "default", cluster: localCluster}
+	}
+
```

Comment:
+1 to @pcn suggestion of logging warn here if len lookup == 0 but allowing boot to continue. can also log info with num clients in success case.